platforms/documentation/docs/src/docs/release/notes.md

```diff
@@ -47,6 +47,46 @@ Note that certain features that rely on third-party tools, such as [PMD](https:/
 
 For details, see the full [compatibility documentation](userguide/compatibility.html#java).
 
+<a name="java-compiler-error-rendering"></a>
+### Java compiler errors in the failure report
+
+From now on, Gradle also includes Java compilation failures in the report at the end of the build.
+This will accompany the already existing errors reported when the task is executed.
+This improvement helps to identify compilation problems faster; with long executions, it can be hard to find the root cause of a compilation failure.
+
+The following is a sample output caused by a simple compilation failure:
+```
+* What went wrong:
+Execution failed for task ':project1:compileJava'.
+> Compilation failed; see the compiler output below.
+
+Java compilation error (compilation:java:java-compilation-error)
+  sample-project/src/main/java/Problem.java:6: error: incompatible types: int cannot be converted to String
+          String a = 1;
+                     ^
+```
+
+If any warning happens during the compilation, it will also be included in the report:
+```
+* What went wrong:
+Execution failed for task ':project1:compileJava'.
+> Compilation failed; see the compiler output below.
+
+Java compilation warning (compilation:java:java-compilation-warning)
```

Comment:
I'm not sure if we discussed this deeply; but I don't think we should have this line in the output. The ID (`compilation:java:java-compilation-warning`) is an internal detail for the problems API and the display name (`Java compilation warning`) contains information already present in the next line.

    sample-project/src/main/java/Problem1.java:6: warning: [cast] redundant cast to String
                                                  ^

Reply:
I understand your concern, but I don't find the discussion about this on the PR, and you didn't block the PR neither.

I can still fix this, just noting that this misunderstanding could've been avoided.